src/Task.ts

```diff
@@ -263,18 +266,32 @@ export class Task {
         return li;
     }
 
-    public toString(): string {
-        const recurrenceRule: string = this.recurrenceRule
-            ? ` 🔁 ${this.recurrenceRule.toText()}`
-            : '';
-        const dueDate: string = this.dueDate
-            ? ` 📅 ${this.dueDate.format(Task.dateFormat)}`
-            : '';
-        const doneDate: string = this.doneDate
-            ? ` ✅ ${this.doneDate.format(Task.dateFormat)}`
-            : '';
-
-        return `${this.description}${recurrenceRule}${dueDate}${doneDate}${this.blockLink}`;
+    public toString(layoutOptions?: LayoutOptions): string {
+
+        var taskString = this.description;
+
+        if (layoutOptions == undefined || !layoutOptions.hideRecurrenceRule) {
```

Comment:
To simplify all the `if`s, you could add something like `layoutOptions = layoutOptions ?? new LayoutOptions()` at the beginning of the function. That way, `layoutOptions` will always be defined.

(`??` is a typescript operator; it relates to JavaScript’s horrible way of handling `null` and `undefined` …)